lib/foreman/gettext/support.rb

```diff
@@ -42,19 +42,31 @@ def self.add_text_domain(locale_domain, locale_dir)
           :report_warning => false
       end
 
-      def self.register_human_localenames
-        original_locale = FastGettext.locale
-        FastGettext.class.class_eval { attr_accessor :human_available_locales }
-        FastGettext.human_available_locales = []
-        FastGettext.default_available_locales.sort.each do |locale|
-          FastGettext.locale = locale
-          # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
-          human_locale = _("locale_name")
-          human_locale = locale if human_locale == "locale_name"
-          FastGettext.human_available_locales << [human_locale, locale]
+      # Loading all available locales to get the language translation is slow
+      # (2 seconds for 20 languages), so let's lazy load that.
+      FastGettext.class.class_eval do
+        @@load_mutex = Mutex.new
+        @@human_available_locales = nil
+
+        def human_available_locales
+          return @@human_available_locales if @@human_available_locales
+          Rails.logger.debug "Loading and caching all available locales"
+          original_locale = FastGettext.locale
+          @@load_mutex.synchronize do
+            @@human_available_locales = []
+            FastGettext.default_available_locales.sort.each do |locale|
+              FastGettext.locale = locale
+              # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
+              human_locale = _("locale_name")
+              human_locale = locale if human_locale == "locale_name"
+              @@human_available_locales << [human_locale, locale]
+            end
+            @@human_available_locales
+          end
+        ensure
+          FastGettext.locale = original_locale
+          @@human_available_locales
```

Comment:
Afaik the value of last expression gets discarded so this doesn't serve any real purpose.

Reply:
I was not sure so went the safest path, will delete then.